code_snippets/go/develop/state.go

```diff
@@ -0,0 +1,48 @@
+package develop
+
+import (
+	restate "github.com/restatedev/sdk-go"
+)
+
+type State struct{}
+
+func (State) Greet(ctx restate.ObjectContext, greeting string) error {
+	// <start_statekeys>
+	stateKeys, err := restate.Keys(ctx)
+	if err != nil {
+		return err
+	}
+	// <end_statekeys>
+	_ = stateKeys
+
+	// <start_get>
+	myString, err := restate.Get[string](ctx, "my-string-key")
+	if err != nil {
+		return err
+	}
+	if myString == "" {
+		myString = "my-default"
+	}
```

Comment:
To make this case abundantly clear, i would use a pointer type instead.

Reply:
I thought about this. But then you have to assign the default to a var and then take a reference to it, which is ugly. This is why people use proto.String. I will take another look